lib/api/api.go

```diff
@@ -1017,16 +1017,9 @@ func (s *service) getSystemStatus(w http.ResponseWriter, r *http.Request) {
 	res["tilde"] = tilde
 	if s.cfg.Options().LocalAnnEnabled || s.cfg.Options().GlobalAnnEnabled {
 		res["discoveryEnabled"] = true
-		discoErrors := make(map[string]string)
-		discoMethods := 0
-		for disco, err := range s.discoverer.ChildErrors() {
-			discoMethods++
-			if err != nil {
-				discoErrors[disco] = err.Error()
-			}
-		}
-		res["discoveryMethods"] = discoMethods
-		res["discoveryErrors"] = discoErrors
+		discoStatus := s.discoverer.FinderStatus()
+		res["discoveryMethods"] = len(discoStatus) // DEPRECATED: Redundant, only for backwards compatibility, should be removed.
+		res["discoveryErrors"] = discoStatus
```

Comment:
I'd add a "discoveryStatus" key and mark "discoveryErrors" as deprecated too, as the errors bit is misleading (I assumed it was only discovery methods with errors at first).

Reply:
This comes down to breaking the API or not.  I chose the path of least breakage when you said either would be okay.  Just trying not to increase the required effort for wrapper developers needlessly.  If you're more comfortable with different fields, I'll be happy to clean them up.

That would be only `discoveryStatus` then, replacing both keys.  In that case use child objects vor values as well?

Reply:
I meant adding a duplicate entry that's named `discoveryStatus` and has the same content as `discoveryErrors`, keeping `discoveryErrors` to not break api (but add a deprecation comment to be addressed whenever a v2 happens).

Reply:
> It's not important though, it may also stay as is and one has to realize that it also contains empty errors for discovery methods that run successfully.

Then I'd say let's leave this in "backward-compatible with slight potential hiccup" state and clean up / rename whenever the deprecation notice on `discoveryMethods` will get addressed.

Reply:
Actually I didn't realize the api filtered out nil errors before. In that case, what's the downside of keeping the current `discoveryErrors` for backwards compat and add `discoveryStatus` with all methods? What we do now is not backwards compatible, and might trip api consumers that will now display a discovery as failed, even though it's running.  
Sorry for the confusion.

Reply:
> In that case, what's the downside of keeping the current `discoveryErrors` for backwards compat and add `discoveryStatus` with all methods?

Unnecessary duplication.

So should the new `discoveryStatus` be an dict of objects then, like `connectionServiceStatus`?  That would be easier to extend later on just in case.  Or just a plain string -> error mapping as we have now?

Reply:
Done.